### PR TITLE
NP-978: User access rights

### PR DESCRIPTION
**Pull request by axthosarouris**
Changes:

User access rights: Description of user access rights
Anonymous user rights: Add description for explicitly prohibiting an anonymous user to publish a publication.

**UPDATE**:
According to the discussion I had with @garshol  publication sharing is a non-MVP feature.
Therefore I made the following changes
Last changes: 

1. Tagged features that are not MVP as `@notmvp'
2. Updated the features so that in the current scope (MVP), a USER has basically the same rights as an ANONYMOUS user

### CODE REVIEW

File: publication-commons/src/test/resources/features/UserAccessRights.feature [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    Then READ returns the publication "PubId"
+
+  Scenario: Users can see published material when they list publications
+    Given that DynamoDBPublicationService has a LIST method
+    And that LIST requires a user with non empty username
+    And that LIST requires a publication owner's username
+    When LIST is called on behalf of the "theUser" for the user "theCreator"
+    Then LIST returns all published publications whose owner is the user "theCreator"
+
+  Scenario: Users cannot create a publication
```
[the remark this answers is not in the document]
Comment (axthosarouris) [reply publication-commons/src/test/resources/features/UserAccessRights.feature:41]: The role name is USER. Maybe I should rephrase?

File: publication-commons/src/test/resources/features/UserAccessRights.feature [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    And there is a user with username "theUser"
+    And the user "theUser" is affiliated with the institution "theInstitution"
+    And the user's role is USER
+    And the user "theUser" does not have any other role
+
+
+  Scenario: Users read published material
+    Given a publication with ID "PubId"
+    And the publication "PubId" has status PUBLISHED
+    When READ is called on behalf of the Anonymous user for the publication "PubId"
```
Comment (garshol) [inline publication-commons/src/test/resources/features/UserAccessRights.feature:31]: Update to a User user, not an Anonymous user.

File: publication-commons/src/test/resources/features/UserAccessRights.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (janbj) [reply publication-commons/src/test/resources/features/UserAccessRights.feature]: Is that Anonymous Users?

File: publication-commons/src/test/resources/features/UserAccessRights.feature [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    And the DynamoDBPublicationService has a PUBLISH method
+    And the PUBLISH method requires a user with non empty username
+    And the PUBLISH method requires a non empty publication ID
+    And there is a user with username "theUser"
+    And the user "theUser" is affiliated with the institution "theInstitution"
+    And the user's role is USER
+    And the user "theUser" does not have any other role
+
+  Scenario: USER users read published material
+    Given a publication with ID "PubId"
```
Comment (janbj) [inline publication-commons/src/test/resources/features/UserAccessRights.feature:28]: General comment, I find the pattern of `USER users`, `user "theUser"` and `publication "PubId"` to be more confusing than just using `an Anonymous User` and `the Publication`with capital letter to show that they are entities in the scenario.

**Pull request merged**